core/app/models/spree/shipment.rb

```diff
@@ -344,7 +345,9 @@ def transfer_to_location(variant, quantity, stock_location)
       end
 
       transaction do
-        new_shipment = order.shipments.create!(stock_location: stock_location)
+        new_shipment = order.shipments.create!(
+          stock_location: stock_location,
+          address: self.address)
 
         order.contents.remove(variant, quantity, {shipment: self})
```

Comment:
this isn't your code, but it seems odd that we are doing order contents things here, maybe just grabbing quantity # of inventory units and moving them over?